src/core/solutions/brown_function.cpp

```diff
@@ -0,0 +1,52 @@
+#include <math.h>
+#include <core/solutions/brown_function.h>
+
+namespace dnn_opt
+{
+namespace core
+{
+namespace solutions
+{
+
+brown* brown::make(generator *generator, unsigned int size)
+{
+  auto* result = new brown(generator, size);
+
+  result->init();
+
+  return result;
+}
+
+float brown::calculate_fitness()
+{
+  float result = 0;
+  float* params = get_params();
+
+  solution::calculate_fitness();
+  
+  int length = size();
+
+  for(int i = 0; i < length - 1.0f; i++)
```

Comment:
Here, `length - 1.0f` is an index, hence it has to be integer, not float. Here the float literal is **not** justified.

May be you are a little bit confused. I suggest that you read [this article ](https://www.thoughtco.com/candand-handling-ints-and-floats-958408), specially the section: 

> Things to Watch out for with ints, floats and bools